src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,150 @@
+        return createElement("div",
+            {
+                className: classNames("widget-pagination", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            createElement(ValidateConfigs, {
+                ...this.props as WrapperProps,
+                queryNode: this.state.targetNode,
+                targetListview: this.state.targetListView,
+                validate: !this.state.findingListviewWidget
+            }),
+            this.renderPageButton()
+        );
+    }
+
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    public static setMessageStatus(currentOffSet: number, offSet: number, maxPageSize: number): string {
+        let fromValue = currentOffSet + 1;
+        let toValue = 0;
+        if (maxPageSize === 0) {
+            fromValue = 0;
+        } else if (maxPageSize < offSet || (currentOffSet + offSet) > maxPageSize) {
+            toValue = maxPageSize;
+        } else {
+            toValue = currentOffSet + offSet;
+        }
+
+        return window.mx.ui.translate(
+            "mxui.lib.MxDataSource",
+            "status",
+            [ fromValue, toValue, maxPageSize ]);
+    }
+
+    private renderPageButton(): ReactElement<PaginationProps> | null {
+        if (this.state.validationPassed) {
+            return createElement(Pagination, {
+                maxPageSize: this.state.maxPageSize,
+                offSet: this.state.offSet,
+                onClickAction: this.updateListView,
+                setMessageStatus: PaginationContainer.setMessageStatus,
+                showPageButton: this.state.showPageButton
+            });
+        }
+
+        return null;
+    }
+
+    private validateConfigs() {
+        if (!this.state.validationPassed) {
+            const queryNode = findDOMNode(this) as HTMLElement;
+            const targetNode = ValidateConfigs.findTargetNode(queryNode);
+            let targetListView: ListView | null = null;
+
+            if (targetNode) {
+                targetListView = dijitRegistry.byNode(targetNode);
+                this.setState({ targetNode });
+
+                if (targetListView) {
+                    this.transformListView(targetNode, targetListView);
+                    this.setState({ targetListView });
+                }
+            }
+            const validateMessage = ValidateConfigs.validate({
+                ...this.props as WrapperProps,
+                queryNode: this.state.targetNode,
+                targetListview: this.state.targetListView,
+                validate: !this.state.findingListviewWidget
+            });
+            this.setState({ findingListviewWidget: false, validationPassed: !validateMessage });
+        }
+    }
+
+    private transformListView(targetNode: HTMLElement, listView: ListView) {
+        const buttonNode = targetNode.querySelector(".mx-listview-loadMore") as HTMLButtonElement;
+
+        this.setState({
```

Comment:
Try to always have state updates at the end of a function's execution path.

Reply:
done